src/Microsoft.TestPlatform.CrossPlatEngine/Client/Parallel/ParallelOperationManager.cs

```diff
@@ -2,219 +2,165 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
-using System.Collections;
-using System.Collections.Concurrent;
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
 
+using Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client.Parallel;
 using Microsoft.VisualStudio.TestPlatform.ObjectModel;
-using Microsoft.VisualStudio.TestPlatform.ObjectModel.Engine;
-
-#nullable disable
 
 namespace Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client;
 
 /// <summary>
-/// Abstract class having common parallel manager implementation
+/// Manages work that is done on multiple testhosts in parallel.
 /// </summary>
-internal abstract class ParallelOperationManager<T, TU> : IParallelOperationManager, IDisposable
+internal sealed class ParallelOperationManager<TManager, TEventHandler, TWorkload> : IDisposable
```

Comment:
This class is just completely different, because now it has a clear purpose. It schedules workloads from a given bag of work on appropriate testhosts. This behavior is re-used for discovery and execution as it should be, instead of implementing the same thing twice in each parallel manager.